Remove mantidplot calls from Indirect

sf1919 opened the pull request:
**Description of work.**
Remove outdated calls to Mantidplot and replace with calls to functions within Workbench.

**To test:**

1. Check Systems tests pass
2. Review code

Fixes #32821.


*This does not require release notes* because **it is not user facing work**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

sf1919 committed:
Removed import_mantidplot from BayesQuasi algorithm

sf1919 committed:
Removed import_mantidplot calls from IndirectBayes and tidied file

- removed import_mantidplot calls from file
- added whitespace around operators
- added whitespace after commas

sf1919 committed:
Removed matidplot calls from IndirectMuscat

- removed calls to matidplot and replaced with plotspectrum and pcolormesh as required
- added whitespaces where required to improve formatting
- changed comment indentdation where appropriate to improve formatting
- removed unnecessary imports

sf1919 committed:
Removed mantidplot calls from IndirectNeutron

- removed calls to mantidplotand replaced with plotspectrum and pcolormesh as appropriate
- removed unncessary imports

sf1919 committed:
Removed mantidplot call from PoldiDataAnalysis

- removed mantidplot calls and replaced with plotspectrum
- removed unncessary imports
- added whitespace for formatting where warnings were occurring

sf1919 committed:
Removed matidplot from MoIDyn

- removed calls to mantidplot and replaced with plotspectrum
- tidied code to reduce warnings

sf1919 committed:
Removed matidplot call method from IndirectImport

sf1919 committed:
Put missing import into PoldiDataAnalysis

Added mantid.kernel import as this is required for Direction calls

SilkeSchomann commented:
@sf1919 Should mantidplot also get replaced in scripts/Inelastic/IndirectReductionCommon.py?

sf1919 commented:
Good shout! I missed that one.

sf1919 committed:
Removed matidplot calls from IndirectReductionCommon

- removed mantidplot calls and replaced with plotspectrum and pcolormesh as appropriate
- added new imports
- minor formatting changes

sf1919 committed:
Move import into method

Moved the pcolormesh import directly into the plot_reduction method

sf1919 commented:
IndirectReductionCommon has a number of methods used by algorithms. As a result it is not possible to used the pcolormesh import for the whole of IndirectReductionCommon as it is a qt function and therefore causes a SimpleAPITest failure (see [e63611a](https://github.com/mantidproject/mantid/pull/33559/commits/e63611ad89d32f42bf0ac226ac61244e02a94e49)). 

This can be bypassed by importing directly into the plot_reduction method as this is not currently called by any of the algorithms. This is perhaps why the mantidplot imports were also directly in the method. Whilst this is not a long term solution it does at least resolve issue #32821 .

I have therefore raised an issue regarding this specific problem separately - see #33597

SilkeSchomann reviewed (approved):
Confirmed that mantidplot is not called anymore from Indirect

DavidFair commented on scripts/Inelastic/IndirectReductionCommon.py:
```diff
@@ -933,9 +933,8 @@ def plot_reduction(workspace_name, plot_type):
-        from mantidplot import importMatrixWorkspace
-        plot_workspace = importMatrixWorkspace(workspace_name)
-        plot_workspace.plotGraph2D()
+        from mantidqt.plotting.functions import pcolormesh
```
Nit: all imports should happen at the top of the file, not in-line with the code for maintainability

sf1919 replied:
See comment above....

DavidFair replied:
I missed the previous comment whilst reviewing :facepalm:

DavidFair reviewed (changes requested):
Nice cleanup, one nit where an import needs to be moved
